src/service/ampdoc-impl.js

```diff
@@ -754,16 +754,22 @@ export class AmpDoc {
   }
 
   /**
-   * Allow one tracking iframe for each amp doc. Caller need to handle user
-   * error when registeration returns false
+   * Attempt to register a service that is allowed/required one for each ampdoc.
+   * Caller need to handle user error when registration returns false.
+   *
+   * Please list your service name below:
+   * 'tracking_iframe',
+   * 'linker-created',
+   *
+   * @param {string} name
    * @return {boolean}
    */
-  registerTrackingIframe() {
-    if (this.hasTrackingIframe_) {
-      return false;
+  registerService(name) {
```

Comment:
Nit: I wouldn't call this `service`, to avoid confusion with ampdoc/window services. Maybe "singleton"?

Reply:
Sounds good. Changed to `registerSingleton()`